src/sidebar/util/fetch-config.js

```diff
@@ -19,6 +21,25 @@ function ancestors(window_) {
 }
 
 /**
+ * Returns the global embedder ancestor frame.
+ *
+ * @param {number} levels - Number of ancestors levels to ascend.
+ * @param {Window=} window
+ * @return {Window}
+ */
+function getAncestorFrame(levels, window_ = window) {
+  let ancestorWindow = window_;
+  for (let i = 0; i < levels; i++) {
+    if (ancestorWindow === ancestorWindow.top) {
+      break;
```

Comment:
Hmm, I wonder if this should throw rather than simply stopping if the number of levels is greater than the depth of the frame tree, since this very likely indicates incorrect configuration.